builders/ant.go

```diff
@@ -0,0 +1,222 @@
+package builders
+
+import (
+	"archive/zip"
+	"bufio"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/bmatcuk/doublestar"
+	"github.com/gnewton/jargo"
+	logging "github.com/op/go-logging"
+
+	"github.com/fossas/fossa-cli/module"
+)
+
+var antLogger = logging.MustGetLogger("ant")
+
+// AntBuilder implements build context for SBT builds
+type AntBuilder struct {
+	AntCmd     string
+	AntVersion string
+
+	JavaCmd     string
+	JavaVersion string
+}
+
+// Initialize collects metadata on Java and SBT binaries
+func (builder *AntBuilder) Initialize() error {
+	antLogger.Debugf("Initializing Ant builder...")
+
+	// Set Java context variables
+	javaCmd, javaVersion, err := which("-version", os.Getenv("JAVA_BINARY"), "java")
+	if err != nil {
+		antLogger.Warningf("Could not find Java binary (try setting $JAVA_BINARY): %s", err.Error())
+	}
+	builder.JavaCmd = javaCmd
+	builder.JavaVersion = javaVersion
+
+	// Set Ant context variables
+	antCmd, antVersionOut, err := which("-version", os.Getenv("ANT_BINARY"), "ant")
+	if err != nil {
+		return fmt.Errorf("could not find Ant binary (try setting $ANT_BINARY): %s", err.Error())
+	}
+
+	builder.AntCmd = antCmd
+	antVersionMatchRe := regexp.MustCompile(`version ([0-9]+\.[0-9]+.\w+)`)
+	match := antVersionMatchRe.FindStringSubmatch(antVersionOut)
+
+	if len(match) == 2 {
+		builder.AntVersion = match[1]
+	}
+
+	antLogger.Debugf("Initialized Ant builder: %#v", builder)
+	return nil
+}
+
+// Build is currently not implemented
+func (builder *AntBuilder) Build(m module.Module, force bool) error {
+	return errors.New("Ant builder is not implemented")
+}
+
+// Analyze resolves a lib directory and parses the jars inside
+func (builder *AntBuilder) Analyze(m module.Module, allowUnresolved bool) ([]module.Dependency, error) {
+	antLogger.Debugf("Running Ant analysis: %#v %#v in %s", m, allowUnresolved, m.Dir)
+
+	options := m.Context.(map[string]interface{})
+
+	libdir := options["libdir"].(string)
```

Comment:
Have you tested this? I'm not sure this is actually plumbed through.

Reply:
See: https://github.com/fossas/fossa-cli/pull/134/files#diff-88e008ed0bc6f169079d9bde601e0edcR107